persistent/tests/test_persistence.py

```diff
@@ -166,6 +228,14 @@ def _test():
             inst._p_oid = new_OID
         self.assertRaises(ValueError, _test)
 
+    def test_assign_p_oid_not_in_cache_allowed(self):
+        jar = self._makeJar()
+        inst = self._makeOne()
+        inst._p_jar = jar
+        inst._p_oid = 1 # anything goes
+        inst._p_oid = 42
+        self.assertEqual(inst._p_oid, 42)
```

Comment:
I interpreted his comments as meaning that the `persistent.Persistent` machinery is *not* tied directly to ZODB, so why impose a ZODB specific requirement on it? Doing so will make it much less useful to others. (True, the picklecache has that requirement, but it's arguable that the picklecache is ZODB or storage specific.)

Documentation aside, the only complete implementation up to this point has allowed arbitrary values for `_p_oid`. De-facto, the documentation is incorrect, and it's a breaking change to enforce specific types of values. I know we have it in our code base, and he has it in his. There are probably others that do too.